spec/lib/hyperion/logger_spec.rb

```diff
@@ -27,16 +25,85 @@
   end
 
   it 'respects the log level' do
-    output = StringIO.new
-    with_stdout(output) do
+    output = capture_stdout do
       Logatron.level = Logatron::ERROR
       logger.debug 'xyzzy'
       logger.error 'qwerty'
       Logatron.level = Logatron::DEBUG
     end
-    output_str = output.string
-    expect(output_str).to include 'qwert'
-    expect(output_str).to_not include 'xyzzy'
+    expect(output).to include 'qwert'
+    expect(output).to_not include 'xyzzy'
+  end
+
+  context '::log_result' do
+    let(:response_desc) { ResponseDescriptor.new('type', 1, :json) }
+    let(:payload_desc) { PayloadDescriptor.new(:protobuf) }
+    let(:route) { RestRoute.new(:get, 'http://test.com', response_desc, payload_desc) }
+
+    context 'for a successful response' do
+      let(:result) { HyperionResult.new(route, HyperionStatus::SUCCESS, 200, 'test') }
+      it 'does not log anything' do
+        output = capture_stdout do
+          log_result(result)
+        end
+        expect(output).to be_empty
+      end
+    end
+
+    context 'for an unsuccessful response' do
+      context 'with no body' do
+        let(:result) { HyperionResult.new(route, HyperionStatus::TIMED_OUT) }
+        it 'does not log anything ' do
+          output = capture_stdout do
+            log_result(result)
+          end
+          expect(output).to be_empty
+        end
+      end
+
+      context 'with a ClientErrorResponse body' do
+        let(:message) { 'test' }
+        let(:code) { ClientErrorCode::MISSING }
+        let(:error) { ClientErrorDetail.new(code, 'resource', field: 'field', value: 1, reason: 'oops') }
+        let(:content) { 'content' }
+        let(:error_response) { ClientErrorResponse.new(message, [error], code, content) }
+        let(:result) { HyperionResult.new(route, HyperionStatus::CLIENT_ERROR, 400, error_response) }
+        it 'logs the key value pairs of the ClientErrorResponse' do
+          output = capture_stdout do
+            log_result(result)
+          end
+          pairs = error_response.as_json
+          errors = pairs.delete('errors').first
+          ensure_key_value_pairs_logged(pairs, output)
+          ensure_key_value_pairs_logged(errors, output)
+        end
+
+        def ensure_key_value_pairs_logged(hash, output)
+          hash.each do |k,v|
+            [k, v].each { |x| expect(output).to include x.to_s }
```

Comment:
Why test for k and v separately rather than `"#{k}=#{v}"`? Trying to prevent brittle tests?